lib/components/ms-modal/MsCodeValidationModal.vue

```diff
@@ -0,0 +1,238 @@
+<template>
+  <ms-modal
+    title="codeVerification.title"
+    :subtitle="{ key: 'codeVerification.subtitle', data: { email: email } }"
+    class="coucou"
+  >
+    <div class="code-input">
+      <template
+        v-for="i in CODE_LENGTH"
+        :key="`code${i - 1}`"
+      >
+        <ion-input
+          ref="inputs"
+          v-model="codes[i - 1]"
+          class="code-input__item title-h1-xl"
+          :class="{ 'has-values': codes[i - 1] !== '' }"
+          type="text"
+          inputmode="numeric"
+          :maxlength="1"
+          @keydown="onKeydown($event)"
+          @ion-input="onIonInput($event)"
+          @keyup.backspace="onKeyUpBackspace($event)"
+          @paste="onPaste($event)"
+        />
+      </template>
+    </div>
+    <div class="code-footer">
+      <div class="bottomtext body">
+        {{ $msTranslate('codeVerification.codeNotReceivedQuestion') }}
+      </div>
+      <div class="bottomtext body">
+        {{ $msTranslate('codeVerification.bottomOptions') }}
+      </div>
+    </div>
+  </ms-modal>
+</template>
+
+<script setup lang="ts">
+import MsModal from '@lib/components/ms-modal/MsModal.vue';
+import { computed, nextTick, onMounted, ref } from 'vue';
+import { IonInput } from '@ionic/vue';
+
+type HTMLIonInputElementOrUndefined = HTMLIonInputElement | undefined;
+const inputs = ref();
+const codes = ref<string[]>([]);
+const CODE_LENGTH = 6;
+const isCodeValid = computed(() => isValidCode(codes.value.join('')));
+
+onMounted(async (): Promise<void> => {
+  await nextTick();
+  focusInputElement(getFirstInputElement());
+});
+
+const props = defineProps<{
```

Comment:
Don't we want to be able to set a custom length?

Reply:
I might be a good idea.